src/handlers.cpp

```diff
@@ -394,19 +406,13 @@ void ImpiTask::on_mar_response(Diameter::Message& rsp)
       send_http_reply(HTTP_SERVER_UNAVAILABLE);
       break;
       // LCOV_EXCL_STOP
-    case 5001:
-    {
-      TRC_INFO("Multimedia-Auth answer with result code %d - reject", result_code);
-      SAS::Event event(this->trail(), SASEvent::NO_AV_HSS, 0);
-      SAS::report_event(event);
-      send_http_reply(HTTP_NOT_FOUND);
-    }
-    break;
     default:
-      TRC_INFO("Multimedia-Auth answer with result code %d - reject", result_code);
+      TRC_INFO("Multimedia-Auth answer with result code %d and experimental result code %d and vendor id %d - reject",
+               result_code, experimental_result_code, vendor_id);
       SAS::Event event(this->trail(), SASEvent::NO_AV_HSS, 0);
       SAS::report_event(event);
-      send_http_reply(HTTP_SERVER_ERROR);
+      send_http_reply((experimental_result_code == DIAMETER_ERROR_USER_UNKNOWN &&
```

Comment:
Can we have an if...else if...else if...else sequence, rather than switching on result code and using a ternary operator in the default case to account for experimental result code?